### PR TITLE
Add field to enable EBS Volume Optimization

### LINKED ISSUE #1808: Possibility to activate EBS-optimized flag on nodes
Is there a way to activate "EBS-optimized" flag on the worker nodes?

I've started m4.2xlarge instances which should be [EBS-optimized by default](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSOptimized.html), but kops gives me instances with EBS-optimized set to false.

### PR DESCRIPTION
**Pull request by austinmoore-**
Fixes #1808

I'm not sure if similar remote storage optimization capabilities are offered on other platforms like GCP. If similar optimizations are offered, then we should open another issue to add support for this field on each platform.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2822)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

File: docs/instance_groups.md [hunk 1]
```diff
@@ -186,3 +186,12 @@ Example: `kops delete ig morenodes`
 
 No rolling-update is needed (and note this is not currently graceful, so there may be interruptions to
 workloads where the pods are running on those nodes).
+
+## EBS Volume Optimization
+
+EBS-Optimized instances can be created by setting the following field:
```
Comment (chrislovecnm) [inline docs/instance_groups.md]: When should we do this?  What instances allow it?  Should we be automatically doing this?

File: pkg/apis/kops/instancegroup.go [hunk 2]
```diff
@@ -76,6 +76,8 @@ type InstanceGroupSpec struct {
 	RootVolumeSize *int32 `json:"rootVolumeSize,omitempty"`
 	// RootVolumeType is the type of the EBS root volume to use (e.g. gp2)
 	RootVolumeType *string `json:"rootVolumeType,omitempty"`
+	// RootVolumeOptimization enables EBS optimization for an instance
+	RootVolumeOptimization *bool `json:"rootVolumeOptimization,omitempty"`
```
Comment (chrislovecnm) [inline pkg/apis/kops/instancegroup.go]: Looks like we need a RootVolume struct ... out of scope of this PR :)

File: upup/pkg/fi/cloudup/awstasks/launchconfiguration.go [hunk 3]
```diff
@@ -49,6 +49,8 @@ type LaunchConfiguration struct {
 	RootVolumeSize *int64
 	// RootVolumeType is the type of the EBS root volume to use (e.g. gp2)
 	RootVolumeType *string
+	// RootVolumeOptimization enables EBS optimization for an instance
+	RootVolumeOptimization *bool `json:"rootVolumeOptimization,omitempty"`
```
Comment (chrislovecnm) [inline upup/pkg/fi/cloudup/awstasks/launchconfiguration.go]: Are you using the JSON serialization for anything?

File: docs/instance_groups.md (hunk 1 above)
Comment (austinmoore-) [inline docs/instance_groups.md]: There are some instances types that cannot be created as EBS-Optimized instances. I created a cluster with an instance type that cannot be optimized, with the `rootVolumeOptimization` flag and this error was thrown by the task:

```
W0627 11:35:51.822386   28707 executor.go:109] error running task "LaunchConfiguration/master-us-east-1b.masters.test.k8s.us-east-1.example.com" (9m48s remaining to succeed): error creating AutoscalingLaunchConfiguration: ValidationError: Should specify either ephemeral volume, EBS volume, or no-device./dev/sda1.
```

We could maintain a list of the instances that cannot be EBS-Optimized instances, and throw a clearer error sooner. My concern is that the list gets out of sync when new instances types are released. But, it does appear that all the new generation instance types (except for T2) support EBS-Optimization, so maybe this list won't change much.

List of instance types that are not supported:

```
m1.small
m1.medium
c1.medium
cc2.8xlarge
cg1.4xlarge
m2.xlarge
cr1.8xlarge
i2.8xlarge
hi1.4xlarge
hs1.8xlarge
t1.micro
t2.nano
t2.micro
t2.small
t2.medium
t2.large
t2.xlarge
t2.2xlarge
m3.medium
m3.large
c3.large
c3.8xlarge
g2.8xlarge
r3.large
r3.8xlarge
```

File: pkg/model/awsmodel/autoscalinggroup_test.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		Tasks: make(map[string]fi.Task),
+	}
+
+	b.Build(c)
+
+	lc := c.Tasks["LaunchConfiguration/nodes.testcluster.test.com"].(*awstasks.LaunchConfiguration)
+
+	if *lc.RootVolumeOptimization == false {
+		t.Fatalf("RootVolumeOptimization was expected to be true, but was false")
+	}
```
Comment (justinsb) [inline pkg/model/awsmodel/autoscalinggroup_test.go]: This is great, and we should keep it.  But FYI you can also achieve a lot of this by using one of the terraform-output tests, for example `./tests/integration/complex/kubernetes.tf`

File: pkg/apis/kops/instancegroup.go (hunk 2 above)
Comment (justinsb) [inline pkg/apis/kops/instancegroup.go]: Agreed - we can put a TODO in there to rationalize in the next API version.  I think also now that local volumes are coming, people will soon be asking for non-root EBS attachments also.

**Pull request merged**